lib/modules/dosomething/dosomething_rogue/dosomething_rogue.module

```diff
@@ -258,6 +147,129 @@ function dosomething_rogue_format_activity($response)
 }
 
 /**
+  * Format a response from Rogue's GET /activity to match
+  * the response from Phoenix's GET /signups/:sid
+  *
+  * @param array $signup
+  */
+function dosomething_rogue_format_signup($signup) {
+  // Get campaign and run information for the signup data response
+  $northstar_user = dosomething_northstar_get_user($signup['northstar_id']);
+
+  // Get campaign information
+  $drupal_user = dosomething_user_get_user_by_northstar_id($signup['northstar_id']);
+
+  try {
+    $campaign_run = dosomething_helpers_get_current_campaign_run_for_user($signup['campaign_id'], $drupal_user); // get errors when calling this sometimes (500 - no campaign data found)
+  } catch(Exception $e) {
+    // the regular Phoenix endpoint doesn't seem to handle exceptions like these so we just set the run to null if there is an exception
+    $campaign_run = null;
+  }
+
+  $campaign_run_nid = $campaign_run->nid;
+
+  $current = ($signups['campaign_run_id'] == $campaign_run_nid) ? true : false;
+
+  if (isset($campaign_run->field_run_date['en'][0])) {
+    $current_run_start = $campaign_run->field_run_date['en'][0]['value'];
+    $current_run_end = $campaign_run->field_run_date['en'][0]['value2'];
+  }
+  else {
+    $current_run_start = NULL;
+    $current_run_end = NULL;
+  }
+
+  $campaign = Campaign::get($signup['campaign_id']);
```

Comment:
Is `$signup` in fact, just one signup array? Seems like there might be an issue where when you are grabbing just one signup via `GET /signups/:sid` it works because `$signup` is just one signup? but when you are trying to grab a bunch of signups, maybe you are not passing the right thing.